Content.Client/UserInterface/Controls/RadialContainer.cs

```diff
@@ -67,36 +67,86 @@ public RadialContainer()
+		const float baseRadius = 100f;
         const float radiusIncrement = 5f;
 		
-        var children = ReserveSpaceForHiddenChildren ? Children : Children.Where(x => x.Visible);
-        var childCount = children.Count();
+        var children = ReserveSpaceForHiddenChildren
+            ? Children
+            : Children.Where(x => x.Visible);
+
+        // elements are children (buttons) and dividers, they all are spread evenly
+        var elementCount = children.Count() * 2;
 		
 		// Add padding from the center at higher child counts so they don't overlap.
-		Radius = baseRadius + (childCount * radiusIncrement);
+		Radius = baseRadius + (elementCount * radiusIncrement);
+
+        var isAntiClockwise = RadialAlignment == RAlignment.AntiClockwise;
 
         // Determine the size of the arc, accounting for clockwise and anti-clockwise arrangements
         var arc = AngularRange.Y - AngularRange.X;
-        arc = (arc < 0) ? MathF.Tau + arc : arc;
-        arc = (RadialAlignment == RAlignment.AntiClockwise) ? MathF.Tau - arc : arc;
+        arc = arc < 0
+            ? MathF.Tau + arc
+            : arc;
+        arc = isAntiClockwise
+            ? MathF.Tau - arc
+            : arc;
 
         // Account for both circular arrangements and arc-based arrangements
-        var childMod = MathHelper.CloseTo(arc, MathF.Tau, 0.01f) ? 0 : 1;
+        var childMod = MathHelper.CloseTo(arc, MathF.Tau, 0.01f)
+            ? 0
+            : 1;
 
         // Determine the separation between child elements
-        var sepAngle = arc / (childCount - childMod);
-        sepAngle *= (RadialAlignment == RAlignment.AntiClockwise) ? -1f : 1f;
+        var sepAngle = arc / (elementCount - childMod);
+        sepAngle *= isAntiClockwise
+            ? -1f
+            : 1f;
+
+        var halfWidth = Width / 2f;
+        var halfHeight = Height / 2f;
 
         // Adjust the positions of all the child elements
-        foreach (var (i, child) in children.Select((x, i) => (i, x)))
+        var query = children.Select((x, index) => (index, x));
+
+        foreach (var (index, child) in query)
         {
-            var position = new Vector2(Radius * MathF.Sin(AngularRange.X + sepAngle * i) + Width / 2f - child.Width / 2f, -Radius * MathF.Cos(AngularRange.X + sepAngle * i) + Height / 2f - child.Height / 2f);
+            // odd indexes are child elements (buttons)
+            var indexForChildElement = index * 2 + 1;
+
+            var targetAngleOfChild = AngularRange.X + sepAngle * indexForChildElement;
+            var childX = Radius * MathF.Sin(targetAngleOfChild) + halfWidth - child.Width / 2f;
+            var childY = -Radius * MathF.Cos(targetAngleOfChild) + halfHeight - child.Height / 2f;
+            var position = new Vector2(childX, childY);
+
             SetPosition(child, position);
+
+            // radial menu buttons need to know in which sector around container they should be rendered
+            if (child is RadialMenuTextureButton tb)
```

Comment:
the best idea i can imagine to counter that - is to creat interface. It have 95% chance to not ever be used, but architectural remark is correct. So it will be done.